### ISSUE TITLE
comment inside of empty enum variant payload with explicit discriminant results in invalid syntax after rustfmt run

### ISSUE THREAD

**Issue by LizzyFleckenstein03** [rust-lang/rustfmt#5686]
Example code:
```rs
#[repr(u8)]
enum MyEnum {
    MyVariant {
        /* Comment */
    } = 0,
}
```
This compiles fine.

rustfmt 1.5.2-nightly (50d3ba5 2023-02-04) inserts a line:
```rs
#[repr(u8)]
enum MyEnum {
    MyVariant {
        /* Comment */
        } =
    } = 0,
}
```

This is a syntax error:
```
error: unexpected closing delimiter: `}`
 --> src/main.rs:7:1
  |
3 |     MyVariant {
  |               - this delimiter might not be properly closed...
4 |         /* Comment */
5 |         } =
  |         - ...as it matches this but it has different indentation
6 |     } = 0,
7 | }
  | ^ unexpected closing delimiter
```

**Comment by calebcartwright** [comment]
Thank you for the report!

For anyone interested in working on this, these types of bugs are almost always caused by an incorrect span derivation that only becomes problematic in the presence of comments. In this case it's an empty struct so I'd suggest taking a look in this general vicinity:

https://github.com/rust-lang/rustfmt/blob/5391847ea5a12262704657a0708b1674fb344403/src/items.rs#L1328-L1332

I suspect that the issue is with the construction of that inner span, and the `hi` value used (the second arg to the `mk_sp` call) should instead be the hi bytepos from before the last `}`

**Comment by ytmimi** [comment]
@Swiiz thanks for reaching out! That's definitely a bug.

Could you please open up a new issue for that case. Although the result is similar the error is occurring in a different context (struct generics vs enum explicit discriminant) and therefore is a distinct issue.

**Comment by ytmimi** [comment]
@Swiiz following up here. The issue that you pointed out is a duplicate of #5358, and [the fix was released in rustfmt v1.5.2](https://github.com/rust-lang/rustfmt/blob/master/CHANGELOG.md#fixed).
